src/views/catalog/wizard/tabs/disks/hooks/useEditDiskState.ts

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+
+import { V1VirtualMachine, V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import {
+  DYNAMIC,
+  OTHER,
+  sourceTypes,
+  volumeTypes,
+} from '@kubevirt-utils/components/DiskModal/DiskFormFields/utils/constants';
+import {
+  DiskFormState,
+  DiskSourceState,
+  initialStateDiskSource,
+} from '@kubevirt-utils/components/DiskModal/state/initialState';
+import { getDisks, getVolumes } from '@kubevirt-utils/resources/vm';
+import { diskTypes } from '@kubevirt-utils/resources/vm/utils/disk/constants';
+import { getDiskDrive, getDiskInterface } from '@kubevirt-utils/resources/vm/utils/disk/selectors';
+
+type UseEditDiskStates = (
+  vm: V1VirtualMachine,
+  diskName: string,
+  vmi?: V1VirtualMachineInstance,
+) => {
+  initialDiskState: DiskFormState;
+  initialDiskSourceState: DiskSourceState;
+};
+
+export const useEditDiskStates: UseEditDiskStates = (vm, diskName) => {
+  const initialDiskSourceState = React.useMemo(() => ({ ...initialStateDiskSource }), []);
+  const disks = getDisks(vm);
+  const disk = disks?.find(({ name }) => name === diskName);
+
+  const { diskSource, diskSize } = React.useMemo(() => {
+    const volumes = getVolumes(vm);
+    const volume = volumes?.find(({ name }) => name === diskName);
+    // volume consists of 2 keys:
+    // name and one of: containerDisk/cloudInitNoCloud
+    const volumeSource = Object.keys(volume).find((key) =>
+      [
+        volumeTypes.CONTAINER_DISK,
+        volumeTypes.DATA_VOLUME,
+        volumeTypes.PERSISTENT_VOLUME_CLAIM,
+      ].includes(key),
+    );
+
+    if (volumeSource === sourceTypes.EPHEMERAL) {
+      initialDiskSourceState.ephemeralSource = volume.containerDisk?.image;
+      return { diskSource: sourceTypes.EPHEMERAL, diskSize: DYNAMIC };
+    }
+    return { diskSource: OTHER, diskSize: null };
```

Comment:
why is diskSize null here?
we need different logic here, we're in the Wizard.
The VM is not yet created, we can edit disks, volumes with much more freedom